deepchecks/nlp/utils/text_properties.py

```diff
@@ -0,0 +1,162 @@
+
+    As identifying the sentiment of a text requires the textblob package, which is not necessary for the rest of the
+    NLP module, we import it only when needed.
+    Therefore, the sentiment property is not given directly, but rather through this function, so that the user will
+    only get an ImportError if they try to use the sentiment property.
+    """
+    try:
+        import textblob  # pylint: disable=import-outside-toplevel
+    except ImportError as e:
+        raise ImportError(
+            'property sentiment requires the textblob python package. '
+            'To get it, run "pip install textblob".') from e
+
+    def sentiment(raw_text: Sequence[str]) -> List[float]:
+        """Return list of floats of sentiment."""
+        return [textblob.TextBlob(text).sentiment.polarity for text in raw_text]
+
+    return sentiment
+
+
+def get_subjectivity_detection() -> callable:
+    """Return a function that returns the subjectivity of a text.
+
+    As identifying the subjectivity of a text requires the textblob package, which is not necessary for the rest of the
+    NLP module, we import it only when needed.
+    Therefore, the subjectivity property is not given directly, but rather through this function, so that the user will
+    only get an ImportError if they try to use the subjectivity property.
+    """
+    try:
+        import textblob  # pylint: disable=import-outside-toplevel
+    except ImportError as e:
+        raise ImportError(
+            'property subjectivity requires the textblob python package. '
+            'To get it, run "pip install textblob".') from e
+
+    def subjectivity(raw_text: Sequence[str]) -> List[float]:
+        """Return list of floats of subjectivity."""
+        return [textblob.TextBlob(text).sentiment.subjectivity for text in raw_text]
+
+    return subjectivity
+
+
+def _get_default_properties(include_properties: List[str] = None, ignore_properties: List[str] = None):
+    """Return the default properties.
+
+    Default properties are defined here and not outside the function so not to import all the packages
+    if they are not needed.
+    """
+    ret = [
+        {'name': 'text_length', 'method': text_length, 'output_type': 'numeric'},
+        {'name': 'average_word_length', 'method': average_word_length, 'output_type': 'numeric'},
+        {'name': 'percentage_special_characters', 'method': percentage_special_characters, 'output_type': 'numeric'},
+    ]
+
+    # Filter by properties or ignore_properties:
+    if include_properties is not None and ignore_properties is not None:
+        raise ValueError('Cannot use properties and ignore_properties parameters together.')
+    elif include_properties is not None:
+        ret = [prop for prop in ret if prop['name'] in include_properties]
+    elif ignore_properties is not None:
+        ret = [prop for prop in ret if prop['name'] not in ignore_properties]
+
+    # Add properties that require additional packages:
```

Comment:
Not sure why need special code for these properties. Can you explain?

Reply:
These are the properties that use outside packages that I didn't want to require users to use. That's why their functions return a function, and need this special treatment in order for the imports to be used only when actually calculating the properties